burn-import/src/pytorch/converter.rs

```diff
@@ -0,0 +1,250 @@
+        self.inputs.push(Path::new(input).into());
+        self
+    }
+
+    /// Set development mode.
+    ///
+    /// If this is set to true, the generated model will be saved as `.graph.txt` files and model
+    /// states will be saved as `.json` file.
+    pub fn development(&mut self, development: bool) -> &mut Self {
+        self.development = development;
+        self
+    }
+
+    /// Run code generation.
+    ///
+    /// This function is intended to be called from `build.rs` script.
+    pub fn run_from_script(&mut self) {
+        self.run(true);
+    }
+
+    /// Run code generation.
+    ///
+    /// This function is intended to be called from CLI.
+    pub fn run_from_cli(&mut self) {
+        self.run(false);
+    }
+
+    /// Specify parameter precision to be saved.
+    ///
+    /// # Arguments
+    ///
+    /// * `half_precision` - If true, half precision is saved. Otherwise, full precision is saved.
+    pub fn half_precision(&mut self, half_precision: bool) -> &mut Self {
+        self.half_precision = half_precision;
+        self
+    }
+
+    /// Specify the type of the record to be saved.
+    ///
+    /// # Arguments
+    ///
+    /// * `record_type` - The type of the record to be saved.
+    pub fn record_type(&mut self, record_type: RecordType) -> &mut Self {
+        self.record_type = record_type;
+        self
+    }
+
+    /// Guess module type using heuristics.
+    pub fn guess_module_type(&mut self, guess_module_type: bool) -> &mut Self {
+        self.guess_module_type = guess_module_type;
+        self
+    }
+
+    /// Run code generation.
+    fn run(&mut self, is_build_script: bool) {
+        log::info!("Starting to convert Pytorch weights to Burn's format");
+
+        // prepend the out_dir to the cargo_out_dir if this is a build script
+        let out_dir = if is_build_script {
+            let cargo_out_dir = env::var("OUT_DIR").expect("OUT_DIR env is not set");
+            let mut path = PathBuf::from(cargo_out_dir);
+
+            // // Append the out_dir to the cargo_out_dir
+            path.push(self.out_dir.clone().unwrap());
+            path
+        } else {
+            self.out_dir.as_ref().expect("out_dir is not set").clone()
+        };
+
+        log::debug!("Output directory: {:?}", out_dir);
+
+        create_dir_all(&out_dir).unwrap();
+
+        for input in self.inputs.clone().iter() {
+            let file_name = input.file_stem().unwrap();
+            let out_file: PathBuf = out_dir.join(file_name);
+
+            log::info!("Converting {:?}", input);
+            log::debug!("Input file name: {:?}", file_name);
+            log::debug!("Output file: {:?}", out_file);
+
+            self.convert_model(input, out_file);
+        }
+
+        log::info!("Finished converting Pytorch weights to Burn's format");
+    }
+
+    /// Convert model weights from the `input` file and save them to the `out_file`.
+    fn convert_model(&mut self, input: &PathBuf, out_file: PathBuf) {
```

Comment:
Better to pass `&Path` or directly `PathBuf` if we do not use it in other functions